ingenialink/virtual/virtual_drive.py

```diff
@@ -475,6 +485,188 @@ class PlantOpenLoopVoltageToCurrentC(PlantOpenLoopVoltageToCurrentA):
     REGISTER_VALUE = "FBK_CUR_C_VALUE"
 
 
+class VirtualPhasing:
+    """Mock of the phasing calibration.
+
+    Args:
+        drive: Instance of the VirtualDrive.
+    """
+
+    PHASING_MODE_REGISTER = "COMMU_PHASING_MODE"
+    PHASING_CURRENT_REGISTER = "COMMU_PHASING_MAX_CURRENT"
+    PHASING_TIMEOUT_REGISTER = "COMMU_PHASING_TIMEOUT"
+    PHASING_ACCURACY_REGISTER = "COMMU_PHASING_ACCURACY"
+
+    REFERENCE_ANGLE_VALUE_REGISTER = "COMMU_ANGLE_REF_VALUE"
+    COMMUTATION_ANGLE_VALUE_REGISTER = "COMMU_ANGLE_VALUE"
+    REFERENCE_ANGLE_OFFSET_REGISTER = "COMMU_ANGLE_REF_OFFSET"
+    COMMUTATION_ANGLE_OFFSET_REGISTER = "COMMU_ANGLE_OFFSET"
+
+    REFERENCE_FEEDBACK_SENSOR = "COMMU_ANGLE_REF_SENSOR"
+
+    CURRENT_CMD_VALUE_REGISTER = "CL_CUR_CMD_VALUE"
+    CURRENT_ACTUAL_VALUE_REGISTER = "FBK_CUR_MODULE_VALUE"
+
+    SETUP_TIME = 1.0
+    INITIAL_ANGLE = 180.0
+    INITIAL_ANGLE_HALLS = 240.0
+    SAMPLING_RATE = 100
+
+    def __init__(self, drive: "VirtualDrive") -> None:
+        self.drive = drive
+        self.start_time = 0.0
+        self.phased = False
+        self.drive.reg_signals[self.REFERENCE_ANGLE_VALUE_REGISTER] = (
+            np.ones(1) * self.reference_angle_offset
+        )
+        self.drive.reg_time[self.REFERENCE_ANGLE_VALUE_REGISTER] = np.zeros(1)
+        self.drive.reg_signals[self.COMMUTATION_ANGLE_VALUE_REGISTER] = (
+            np.ones(1) * self.commutation_angle_offset
+        )
+        self.drive.reg_time[self.COMMUTATION_ANGLE_VALUE_REGISTER] = np.zeros(1)
+        self.drive.reg_signals[self.CURRENT_CMD_VALUE_REGISTER] = np.zeros(1)
+        self.drive.reg_time[self.CURRENT_CMD_VALUE_REGISTER] = np.zeros(1)
+        self.drive.reg_signals[self.CURRENT_ACTUAL_VALUE_REGISTER] = np.zeros(1)
+        self.drive.reg_time[self.CURRENT_ACTUAL_VALUE_REGISTER] = np.zeros(1)
+        self.drive.reg_noise_amplitude[self.CURRENT_ACTUAL_VALUE_REGISTER] = 0.01
+
+    def enable(self) -> None:
+        """Enable phasing and create signals."""
+        if self.phasing_mode != PhasingMode.FORCED or self.phasing_bit:
```

Comment:
Maybe you could add a logging if this condition is accomplished. Just in case. What do you think?

Reply:
You are not logging anything in the virtual drive. I'll create an issue to log this and other relevant events.